### PR TITLE
(cli) Rename cli fetch & push to read & write

### PR DESCRIPTION
**Pull request by Leobouloc**
## Purpose

As mentionned in #383 , cli currently uses `fetch` and `push`. This could be unified with the new terms `read` and `write`.

## Proposal

- [x] change names in cli
- [x] update tests and doc
- [x] add to changelog

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -25,6 +25,8 @@ General improvement for the Helm Chart:
 - add missing namespace label
 - make object name consistent
 
+- In cli, change `push` to `write` and `fetch` to `read`
```
Comment (quitterie-lcs) [inline CHANGELOG.md:28]: You can just put this bullet point on line 13

File: tests/test_cli.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-"""Tests for Ralph cli."""
+"""Test for Ralph cli."""
```
Comment (quitterie-lcs) [inline tests/test_cli.py:1]:
```suggestion
"""Tests for Ralph cli."""
```

**Review by quitterie-lcs — COMMENTED** [review]
Looks good! Only a few comments and it is ready to be merged !
Be careful that the linter raises W0621 for `write` parameter in auth. A proper way would be to change this parameter name

File: CHANGELOG.md (hunk 1 above)
Comment (Leobouloc) [reply CHANGELOG.md:28]: yes !

File: tests/test_cli.py (hunk 2 above)
Comment (Leobouloc) [reply tests/test_cli.py:1]: indeed

File: CHANGELOG.md [hunk 3]
```diff
@@ -12,6 +12,7 @@ and this project adheres to
 
 ### Changed
 
+- In cli, change `push` to `write` and `fetch` to `read`
```
Comment (jmaupetit) [inline CHANGELOG.md:15]: 1. It's often a good idea to flag breaking changes.
2. Also, maybe we need to prefix change's category, _i.e._ _Helm_ or _CLI_ etc. WDYT?

```suggestion
- CLI: change `push` to `write` and `fetch` to `read` [BC]
```

File: src/ralph/backends/storage/fs.py [hunk 4]
```diff
@@ -62,7 +62,7 @@ def list(self, details=False, new=False):
         logger.debug("Found %d archives", len(archives))
 
         if new:
-            archives = set(archives) - set(self.get_command_history(self.name, "fetch"))
+            archives = set(archives) - set(self.get_command_history(self.name, "read"))
```
Comment (jmaupetit) [inline src/ralph/backends/storage/fs.py:65]: That's tricky. Should we support legacy history?

Comment (jmaupetit) [reply src/ralph/backends/storage/fs.py:65]: Or provide an history migration script.

File: src/ralph/cli.py [hunk 5]
```diff
@@ -314,7 +314,7 @@ def auth(
     username,
     password,
     scope,
-    write,
+    write_credentials,
```
Comment (jmaupetit) [inline src/ralph/cli.py:317]: I don't think it's required.

Comment (Leobouloc) [reply src/ralph/cli.py:317]: since the function is also named `write`, this was done to conform to linter.

File: CHANGELOG.md (hunk 3 above)
Comment (Leobouloc) [reply CHANGELOG.md:15]: I agree

File: src/ralph/cli.py (hunk 5 above)
Comment (jmaupetit) [reply src/ralph/cli.py:317]: What about calling it `write_to_disk`?

File: src/ralph/backends/storage/fs.py (hunk 4 above)
Comment (Leobouloc) [reply src/ralph/backends/storage/fs.py:65]: As decided, we shall provide a migration script in the documentation. 

For memory, the alternative was to provide an additional `migrate` command to cli. Also, it was decided NOT to support legacy.

**Review by wilbrdt — COMMENTED** [review]
Alt text
Looks good, just a couple of remarks before merging it!

File: src/ralph/cli.py [hunk 6]
```diff
@@ -304,7 +304,7 @@ def wrapper(command):
 )
 @click.option(
     "-w",
-    "--write",
+    "--write-credentials",
```
Comment (Leobouloc) [inline src/ralph/cli.py:307]:
```suggestion
    "--write-to-disk",
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -12,6 +12,7 @@ and this project adheres to
 
 ### Changed
 
+- CLI: change `push` to `write` and `fetch` to `read` [BC]
```
[the remark this answers is not in the document]
Comment (Leobouloc) [reply CHANGELOG.md:15]: my bad. will do !

**Review by wilbrdt — APPROVED** [review]
:tada:

**Pull request merged**